### PR TITLE
Fix application.js incorrect overwritten issue

### PR DESCRIPTION
**Pull request by yorzi**
Just comment out incompatible requires. Refers to #140

### CODE REVIEW

File: lib/generators/react_on_rails/base_generator.rb [hunk 1]
```diff
@@ -85,9 +85,9 @@ def update_application_js
 
       def strip_application_js_of_incompatible_sprockets_statements
         application_js = File.join(destination_root, "app/assets/javascripts/application.js")
-        gsub_file(application_js, "//= require jquery_ujs", "")
-        gsub_file(application_js, %r{//= require jquery$}, "")
-        gsub_file(application_js, "//= require_tree .", "")
+        gsub_file(application_js, "//= require jquery_ujs", "// require jquery_ujs")
+        gsub_file(application_js, %r{//= require jquery$}, "// require jquery")
+        gsub_file(application_js, %r{//= require_tree .$}, "// require_tree .")
```
Comment (robwise) [inline lib/generators/react_on_rails/base_generator.rb]: I believe the period needs to be escaped, right?

File: spec/react_on_rails/generators/install_generator_spec.rb [hunk 2]
```diff
@@ -143,7 +143,7 @@
   end
 
   context "without existing application.js or application.js.coffee file" do
-    before(:all) { run_generator_test_with_args([], application_js: false) }
+    before(:all) { run_generator_test_with_args([], application_js: true) }
```
Comment (robwise) [inline spec/react_on_rails/generators/install_generator_spec.rb]: we still want to test for it creating an application_js file if one is missing, so we don't want to set this to true for this test, right?

File: spec/react_on_rails/support/shared_examples/base_generator_examples.rb [hunk 3]
```diff
@@ -48,12 +48,15 @@
     end
   end
 
-  it "removes incompatible sprockets require statements" do
+  it "comments out incompatible sprockets require statements" do
     assert_file("app/assets/javascripts/application.js") do |contents|
-      refute_match("//= require_tree .", contents)
+      refute_match(%r{//= require_tree .$}, contents)
+      assert_match(%r{// require_tree .$}, contents)
```
Comment (robwise) [inline spec/react_on_rails/support/shared_examples/base_generator_examples.rb]: see other comment about escaping the period

File: lib/generators/react_on_rails/base_generator.rb (hunk 1 above)
Comment (yorzi) [inline lib/generators/react_on_rails/base_generator.rb]: Yeah, you right... But it covers the case correctly here when there is a `$` in the ending. Anyway, it's better to code with preciseness. I will fix it.

**Comment by justin808** [comment]
Thanks @yorzi!

**Pull request merged**